lib/archethic/transaction_chain/transaction/validation_stamp/ledger_operations.ex

```diff
@@ -288,15 +288,25 @@ defmodule Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperation
         contract_context \\ nil
       ) do
     # Since AEIP-19 we can consume from minted tokens
-    inputs = inputs ++ tokens_to_mint
+    consolidated_inputs = tokens_to_mint ++ inputs
 
-    %{uco: uco_balance, token: tokens_balance} = ledger_balances(inputs)
+    %{uco: uco_balance, token: tokens_balance} = ledger_balances(consolidated_inputs)
     %{uco: uco_to_spend, token: tokens_to_spend} = total_to_spend(fee, movements)
 
     if sufficient_funds?(uco_balance, uco_to_spend, tokens_balance, tokens_to_spend) do
       consumed_utxos =
-        get_inputs_to_consume(
-          inputs,
+        consolidated_inputs
+        |> Enum.map(fn
+          utxo = %UnspentOutput{from: ^change_address} ->
+            # As the minted tokens are used internally during transaction's validation
+            # and doesn't not exists outside, we use the burning address
+            # to identify inputs coming from the token's minting.
+            %{utxo | from: burning_address()}
```

Comment:
why this choice? shouldn't this be current transaction address? (or a `:mint` from ?)

Reply:
The intent was to differentiate between inputs which are coming from input transaction and some produced dynamically by the transaction (minting)

For instance, on EVM, when you create a contract you send transaction to 0x0000... and the explorer shows the destination as "Contract Creation", we could then adapt this kind of logic.